scaletest/workspacetraffic/run.go

```diff
@@ -186,20 +188,28 @@ type countReadWriter struct {
 	io.ReadWriter
 	bytesRead    atomic.Int64
 	bytesWritten atomic.Int64
+	metrics      *Metrics
+	labels       []string
 }
 
 func (w *countReadWriter) Read(p []byte) (int, error) {
+	start := time.Now()
 	n, err := w.ReadWriter.Read(p)
-	if err == nil {
+	w.metrics.ReadLatencySeconds.WithLabelValues(w.labels...).Observe(time.Since(start).Seconds())
+	if n > 0 {
 		w.bytesRead.Add(int64(n))
+		w.metrics.BytesRead.WithLabelValues(w.labels...).Add(float64(n))
 	}
 	return n, err
 }
 
 func (w *countReadWriter) Write(p []byte) (int, error) {
+	start := time.Now()
 	n, err := w.ReadWriter.Write(p)
-	if err == nil {
+	w.metrics.WriteLatencySeconds.WithLabelValues(w.labels...).Observe(time.Since(start).Seconds())
```

Comment:
I'm curious if we need another metric (bytes?) to calculate the bandwidth. As far as I see, there are only `WriteLatencySeconds` and total `BytesWritten`.  I'm curious if we need a metric to calculate "bytes per write in time"

Reply:
Given that bytes per tick and ticks per second are inputs to this, I think we should probably be OK. But good food for thought!

Reply:
You calc throughput by the `(count at time X) - (count at time Y)`. We should not be doing that math in coderd imo

Reply:
Good catch, will address in a follow-up

Reply:
Prometheus itself will happily compute the time derivative of any metric, so a counter of bytes written is all we need.